src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/UpgradeToGeneratedRegexAnalyzerTests.cs

```diff
@@ -979,6 +979,41 @@ static void Main(string[] args)
             await VerifyCS.VerifyCodeFixAsync(test, expectedFixedCode);
         }
 
+        [Fact]
+        public async Task InterpolatedStringLiteralSyntaxFixedWhenStringLiteralIsConstantField()
+        {
+            string test = @"using System.Text.RegularExpressions;
+
+partial class Program
+{
+    const string pattern = @""a|b\s\n"";
+    const string pattern2 = $""{pattern}2"";
+
+    static void Main(string[] args)
+    {
+        Regex regex = [|new Regex(pattern2)|];
```

Comment:
Do we have a test that validates it does the "right thing" if the const is local to the method instead?

Reply:
If by the right thing, you mean that the local const literal should not be referenced inside GeneratedRegex (because it's out of scope). I think this test does the job :

https://github.com/dotnet/runtime/blob/58d1ecb3e50ead5cd48752bbaedf6dc23ccd2523/src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/UpgradeToGeneratedRegexAnalyzerTests.cs#L948-L979

Reply:
> I think this test does the job

https://github.com/dotnet/runtime/blob/1eed062e48750b60eea5f201ca8c975548b2ad27/src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/UpgradeToGeneratedRegexAnalyzerTests.cs#L975-L976

Isn't that test supposed to generate code that refers to the exact same const? i.e:
```cs
    [GeneratedRegex(pattern2)]
    private static partial Regex MyRegex();
```
@pedrobsaila could you cover this scenario?

Reply:
> Isn't that test supposed to generate code that refers to the exact same const? i.e:

no because pattern2 is local variable, it cant' be referenced outside the method if I'm not mistaken

Reply:
Ah right, then could you fix the double slushes issue and keep @""a|b\s\n""?

Reply:
my bad forgot about that :) , 'I'll do it